src/dxtbx/format/FormatNXmxEigerFilewriter.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import re
+
+import h5py
+
+from scitbx.array_family import flex
+
+import dxtbx.nexus
+from dxtbx.format.FormatNXmx import FormatNXmx
+from dxtbx.nexus import _dataset_as_flex, get_detector_module_slices
+
+DATA_NNNNNN_RE = re.compile(r"data_\d{6}")
+
+
+class FormatNXmxEigerFilewriter(FormatNXmx):
+
+    _cached_file_handle = None
+
+    @staticmethod
+    def understand(image_file):
+        with h5py.File(image_file) as handle:
+            if "/entry/instrument/detector/detectorSpecific/eiger_fw_version" in handle:
+                return True
+        return False
+
+    def __init__(self, image_file, **kwargs):
+        """Initialise the image structure from the given file."""
+        super().__init__(image_file, **kwargs)
+
+    def _get_nxmx(self, fh: h5py.File):
+        nxmx = dxtbx.nexus.nxmx.NXmx(fh)
+        nxentry = nxmx.entries[0]
+
+        nxdetector = nxentry.instruments[0].detectors[0]
+        if nxdetector.underload_value is None:
+            nxdetector.underload_value = 0
+
+        # data_size is reversed - we should probably be more specific in when
+        # we do this, i.e. check data_size is in a list of known reversed
+        # values
+        for module in nxdetector.modules:
+            module.data_size = module.data_size[::-1]
+        return nxmx
+
+    def get_raw_data(self, index):
+        nxmx = self._get_nxmx(self._cached_file_handle)
+        nxdata = nxmx.entries[0].data[0]
+        nxdetector = nxmx.entries[0].instruments[0].detectors[0]
+        raw_data = get_raw_data(nxdata, nxdetector, index)
+        if self._bit_depth_readout:
+            # if 32 bit then it is a signed int, I think if 8, 16 then it is
```

Comment:
I am not sure we should include 8 bit here as I am not sure what else is broken there. 16-wise though your comment is absolutely correct.

Reply:
This is just copied over from the [existing `FormatNXmx` method](https://github.com/cctbx/dxtbx/blob/a46db5c6688c0d801d7cf9a507aeb1bf601dc967/src/dxtbx/format/FormatNXmx.py#L110-L126) so tempted to leave as is for now.